### DISCUSSION TITLE
Failed push

### DISCUSSION THREAD

**Discussion by ghost** [TheAlbertDev/MASB#49] [Q&A] [answered]
Hi Albert, 

I tried to `git push` my final report but it returns the following error: 

![image](https://user-images.githubusercontent.com/78802420/110016793-c0646900-7d25-11eb-96db-5e969b1ec9d5.png)

Earlier today, I commited the report by mistake when it was not finished. I deleted it manually from GitHub. It's possible that this is the origin of the error? I tried `git pull` (as the terminal recommends), but after the following message I'm afraid the branches merge... 
![image](https://user-images.githubusercontent.com/78802420/110017070-133e2080-7d26-11eb-9a2a-8c8bf4568a9e.png)

How can I solve it? Thank you

**Comment by ghost** [comment]
I saw in the "history" of my B-branch that the merge did happen. The thing is, that the merge is between `B-julia` and `B-julia`, it didn't merge with the `master`. 

![image](https://user-images.githubusercontent.com/78802420/110202048-9bc3da80-7e66-11eb-9419-5ddfb378d653.png)

I tried again to commit the report, but I cannot see it in my history although it was (in theory) done: 

![image](https://user-images.githubusercontent.com/78802420/110202080-d9286800-7e66-11eb-92e2-ec28377fd0d0.png)

Is this a serious problem?

**Comment by Albert-Alvarez** [comment, marked as answer]
In principle, there is no problem to merge a branch with itself. That seams (form your first post) that you have started to work from a PC that didn't have the local repository updated with the last changes in the remote repository (maybe you edited something from another PC or from the website directly). When you try to push, Git told you that you missed some updates from the remote repository, and you need to incorporate them to your local repository before push. The terminal told you what command to use (`git pull`). When you do so, a merge commit is performed to incorporate the missed updates from the remote repository.

For me, it looks fine 😉

**Comment by ghost** [comment]
Thank you!!
